sound/soc/sof/control.c

```diff
@@ -114,6 +114,28 @@ int snd_sof_volume_put(struct snd_kcontrol *kcontrol,
 	return change;
 }
 
+int snd_sof_volume_info(struct snd_kcontrol *kcontrol, struct snd_ctl_elem_info *uinfo)
+{
+	struct soc_mixer_control *sm = (struct soc_mixer_control *)kcontrol->private_value;
+	struct snd_sof_control *scontrol = sm->dobj.private;
+	unsigned int channels = scontrol->num_channels;
+	int platform_max;
+
+	if (!sm->platform_max)
+		sm->platform_max = sm->max;
+	platform_max = sm->platform_max;
```

Comment:
It wouldn't hurt to add a comment on what max and platform_max represent? I am not familiar with all this and can't follow if the code makes sense.

Reply:
@plbossart This is verbatim following soc-opc.c:snd_soc_info_volsw_range(), i.e. the function we override. There seems to be support to a range of channels that do not start from 1 and thus the code is a bit funky. But yeah, following soc-ops.c seems fine.